homeassistant/components/upb/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""Support the UPB PIM."""
+import asyncio
+import logging
+
+import upb_lib
+
+from homeassistant.const import CONF_FILE_PATH, CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+
+UPB_PLATFORMS = ["light"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, hass_config: ConfigType) -> bool:
+    """Set up the UPB platform."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up a new config_entry for UPB PIM."""
+
+    url = config_entry.data[CONF_HOST]
+    file = config_entry.data[CONF_FILE_PATH]
+
+    upb = upb_lib.UpbPim({"url": url, "UPStartExportFile": file})
+    upb.connect()
+    hass.data[DOMAIN] = {"upb": upb}
```

Comment:
```suggestion
    hass.data[DOMAIN][config_entry.entry_id] = {"upb": upb}
```

Looks like multiple entries are supported.  The second config entry will overwrite the first one.